src/services/BudgetService.ts

```diff
@@ -183,4 +188,65 @@ export class BudgetService {
       await QuarterBudgetModel.updateBudget(budget)
     }
   }
+
+  static getUncappedRoundedPercentage(value: number, total: number): number {
+    return Math.round((value * 100) / total)
+  }
+
+  static async getExpectedAllocatedBudget() {
+    const activeGrantProposals = await ProposalService.getActiveGrantProposals()
+    const categories = Object.values(NewGrantCategory).map(snakeCase)
+    const currentBudget = await this.getCurrentBudget()
+    const expectedBudget: ExpectedBudget = this.initializeExpectedBudget(categories, currentBudget)
+    this.addContestants(activeGrantProposals, expectedBudget)
+
+    categories.forEach((cat) => {
+      expectedBudget.categories[cat].contested_over_available_percentage = this.getUncappedRoundedPercentage(
+        expectedBudget.categories[cat].contested,
+        expectedBudget.categories[cat].available
+      )
+      expectedBudget.categories[cat].contestants.forEach((contestingGrant: ContestingGrantProposal) => {
+        contestingGrant.contested_percentage = this.getUncappedRoundedPercentage(
+          contestingGrant.size,
+          expectedBudget.categories[cat].contested
+        )
+      })
+    })
+    return expectedBudget
+  }
+
+  private static addContestants(activeGrantProposals: ProposalAttributes[], expectedBudget: ExpectedBudget) {
+    for (const proposal of activeGrantProposals) {
+      const proposalCategory = snakeCase(proposal.configuration.category)
+      const proposalSize = proposal.configuration.size
+
+      expectedBudget.categories[proposalCategory].contested += proposalSize
+      expectedBudget.categories[proposalCategory].contestants.push(this.getContestingGrantProposal(proposal))
+      expectedBudget.total_contested += proposalSize
+    }
+  }
+
+  private static initializeExpectedBudget(categories: string[], currentBudget: CurrentBudget): ExpectedBudget {
```

Comment:
You can also pass `activeGrantProposals` here and move the addContestants code inside this function as part of the expected budget initialization.